File: CPE/a134.00948/main.py
def make_fib():
    fib = [1, 2]
    for i in range(2, 38):
        fib.append(fib[i - 2] + fib[i - 1])
    fib = fib[::-1]
    return fib


fib_lst = make_fib()


def fib_bits(x):
    tmp = x
    bits = ""
    for i in fib_lst:
        if tmp >= i:
            tmp -= i
            bits += "1"
        else:
            bits = bits + "0"
    return bits


def main():
    n = int(input())
    nums = [0] * n
    for i in range(n):
        nums[i] = int(input())

    for i in nums:
        tmp = fib_bits(i).lstrip("0")
        print(f"{i} = {tmp} (fib)")


main()

File: CPE/a134.00948/main.cpp
#include <iostream>
using namespace std;

int main()
{
    int fib_length = 38;
    long long int fib[fib_length] = {0};
    fib[0] = 1;
    fib[1] = 2;

    for (int i = 2; i < fib_length; i++)
    {
        fib[i] = fib[i - 2] + fib[i - 1];
    }

    long long int reverse_fib[fib_length] = {0};

    for (int i = 0; i < fib_length; i++)
    {
        reverse_fib[i] = fib[fib_length - i - 1];
    }

    int n;
    cin >> n;

    int nums[n] = {0};

    for (int i = 0; i < n; i++)
    {
        cin >> nums[i];
    }

    string output = "";
    for (int i = 0; i < n; i++)
    {
        output = "";
        int tmp = nums[i];
        bool flag = true;
        for (int j = 0; j < fib_length; j++)
        {
            if (tmp >= reverse_fib[j])
            {
                flag = false;
                tmp -= reverse_fib[j];
                output += "1";
            }
            else
            {
                if (flag)
                {
                    continue;
                }
                output += "0";
            }
        }
        cout << nums[i] << " = " << output << " (fib)" << endl;
    }

    return 0;
}